src/azimuth/util/prefs.h

```diff
@@ -28,20 +28,44 @@
 
 /*===========================================================================*/
 
-#define AZ_PREFS_UP_KEY_INDEX 0
-#define AZ_PREFS_DOWN_KEY_INDEX 1
-#define AZ_PREFS_RIGHT_KEY_INDEX 2
-#define AZ_PREFS_LEFT_KEY_INDEX 3
-#define AZ_PREFS_FIRE_KEY_INDEX 4
-#define AZ_PREFS_ORDN_KEY_INDEX 5
-#define AZ_PREFS_UTIL_KEY_INDEX 6
-#define AZ_PREFS_PAUSE_KEY_INDEX 7
-#define AZ_PREFS_NUM_KEYS 8
+// The current control that a key press corresponds to.
+typedef enum {
+  AZ_CONTROL_NONE = 0,
+  AZ_CONTROL_UP,
+  AZ_CONTROL_DOWN,
+  AZ_CONTROL_RIGHT,
+  AZ_CONTROL_LEFT,
+  AZ_CONTROL_FIRE,
+  AZ_CONTROL_ORDN,
+  AZ_CONTROL_UTIL,
+  AZ_CONTROL_PAUSE,
+  AZ_CONTROL_BOMBS,
+  AZ_CONTROL_CHARGE,
+  AZ_CONTROL_FREEZE,
+  AZ_CONTROL_TRIPLE,
+  AZ_CONTROL_HOMING,
+  AZ_CONTROL_PHASE,
+  AZ_CONTROL_BURST,
+  AZ_CONTROL_PIERCE,
+  AZ_CONTROL_BEAM,
+  AZ_CONTROL_ROCKETS,
+} az_control_id_t;
+
+#define AZ_FIRST_CONTROL ((int)AZ_CONTROL_UP)
+#define AZ_NUM_CONTROLS ((int)AZ_CONTROL_ROCKETS + 1)
+
+// Manually curated bi-directional map for keys <-> controls:
```

Comment:
An alternative to maintaining a bidirectional map (and trying to avoid breaking invariants by accident) would be to just have the `key_for_control` array, and have an `az_control_for_key()` function that does a linear search over `key_for_control`.  It's still O(1) since `AZ_NUM_CONTROLS` is a (small!) constant, and it'll be super-fast in practice since `key_for_control` fits in a cache line -- and it doesn't matter anyway because it's not a performance-critical function.  Azimuth is chock full of linear scans over fixed-sized arrays like this.  (-:  It seems like it might simplify a bunch of code to do it that way.

Reply:
Done!